pkg/kubectl/sorting_printer.go

```diff
@@ -93,13 +97,20 @@ func SortObjects(decoder runtime.Decoder, objs []runtime.Object, fieldInput stri
 	}
 
 	for ix := range objs {
-		item := objs[ix]
-		switch u := item.(type) {
-		case *runtime.Unknown:
-			var err error
-			if objs[ix], _, err = decoder.Decode(u.RawJSON, nil, nil); err != nil {
+		var err error
+		partialType := strings.Split(reflect.TypeOf(objs[ix]).String(), ".")[0]
+		switch partialType {
+		case "*api":
```

Comment:
this seems odd... why wouldn't we always attempt to convert to the output version for non *runtime.Unknown objects, and in error cases, just use the object as-is for sorting?